### PR TITLE
Data ID Type

### PR DESCRIPTION
**Pull request by grevych**
[OREX-449](https://outreach-io.atlassian.net/browse/OREX-449)

Return and receive data id primary attribute as an integer to match current api v2 implementation.

### CODE REVIEW

File: request.go [hunk 1]
```diff
@@ -494,7 +499,8 @@ func handleTime(attribute interface{}, args []string, fieldValue reflect.Value)
 func handleNumeric(
```
Comment (antonaudzei-outreach) [inline request.go:499]: It seems that we no longer support float IDs, is this function needed?

Comment (grevych) [reply request.go:499]: Yes, attributes still can be of type float. Take a look at:  https://github.com/getoutreach/jsonapi/blob/49369807e985fdf44a9cc1f8a0cd7fe44862149a/request.go#L417

**Comment by gabrielgilini** [comment]
So, do all endpoints in v2 use a numerical for `id`? I checked the [JSON:API specification](https://jsonapi.org/format/#document-resource-object-identification) and it says and id *must* be a string. Looking at our v2 docs, it seems that at least some of them are `int` so it might be the case that the whole v2 is not following the spec in that regard. But I wonder if some endpoints are actually using strings.

**Comment by grevych** [comment]
@gabrielgilini I discussed this with @antonaudzei-outreach

From slack:
> yeah, all of our IDs in APIv2 are integers and that is not JSON-API compliant
>it was brought up by one fresh customers some time ago through support
>our response was - yeah, we know it, but that’s how things are from the beginning so got to stick to it to not break old partners’ code

**Review by antonaudzei-outreach — APPROVED** [review]

**Pull request merged**